src/components/LetterCasePicker.tsx

```diff
@@ -0,0 +1,94 @@
+import React, { FC, memo, useRef } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { titleCase } from 'title-case'
+import { formatLetterCaseActionCreator as formatLetterCase } from '../actions/formatLetterCase'
+import { isTouch } from '../browser'
+import useWindowOverflow from '../hooks/useWindowOverflow'
+import getThoughtById from '../selectors/getThoughtById'
+import themeColors from '../selectors/themeColors'
+import fastClick from '../util/fastClick'
+import head from '../util/head'
+import TriangleDown from './TriangleDown'
+import LowerCaseIcon from './icons/LowerCaseIcon'
+import SentenceCaseIcon from './icons/SentenceCaseIcon'
+import TitleCaseIcon from './icons/TitleCaseIcon'
+import UpperCaseIcon from './icons/UpperCaseIcon'
+
+/** Letter Case Picker component. */
+const LetterCasePicker: FC<{ fontSize: number; style?: React.CSSProperties }> = memo(({ fontSize, style }) => {
+  const colors = useSelector(themeColors)
+  const ref = useRef<HTMLDivElement>(null)
+  const dispatch = useDispatch()
+  const overflow = useWindowOverflow(ref)
+
+  /** Toggles the Letter Case to the clicked swatch. */
+  const toggleLetterCase = (command: string, e: React.MouseEvent | React.TouchEvent) => {
+    e.stopPropagation()
+    e.preventDefault()
+    dispatch(formatLetterCase(command))
+  }
+  const selected = useSelector(state => {
+    const value = (!!state.cursor && getThoughtById(state, head(state.cursor))?.value) || ''
+    if (value === value.toLowerCase()) return 'LowerCase'
+    if (value === value.toUpperCase()) return 'UpperCase'
+    const sentenceCaseRegex = /(^\w|\.\s*\w)/gi
+    if (value === value.toLowerCase().replace(sentenceCaseRegex, match => match.toUpperCase())) return 'SentenceCase'
+    if (value === titleCase(value.toLowerCase())) return 'TitleCase'
+    return ''
+  })
+  const casingTypes = ['LowerCase', 'UpperCase', 'SentenceCase', 'TitleCase']
+
+  return (
+    <div style={{ userSelect: 'none' }}>
+      <div
+        ref={ref}
+        style={{

```

Comment:
This project uses [PandaCSS](https://panda-css.com/) for styling. The `style` attribute should only be used for styles with dynamic runtime values. Otherwise you should use Panda's `css` function. See other components for usage, or check out https://panda-css.com/docs/concepts/writing-styles.